### DISCUSSION TITLE
Things what difficult to create with bubbletea

### DISCUSSION THREAD

**Discussion by antonmedv** [charmbracelet/bubbletea#1055] [General]
Hello,

I've been using bubbletea for a bunch of CLI apps (fx, walk and other), and I love it. It is great. 

But some things very difficult to implement with "string composition" approach. For example like this one: https://github.com/darrenburns/posting

A few things what missing in bubbletea for my perspective:

- components
- forms
- **flex layout** (this is post important, sometimes thinking in term of flex is just easier)
- popovers

What is your thought on this? 

And thanks for the awesome library ❤️

**Comment by meowgorithm** [comment]
Hey, @antonmedv! Thanks for the kind words, and the feedback!

* For components, what types of things are you looking for exactly? (I assume you've seen [Bubbles](https://github.com/charmbracelet/bubbles))
* For forms check out [Huh](https://github.com/charmbracelet/huh)
* Flex is a good one we've love to see official support for; nothing planned in that area yet but the community has [a flexbox implementation](https://github.com/76creates/stickers) for Bubble Tea and Lip Gloss
* Popovers will require some low level work first, which is in the works

**Reply by antonmedv** [reply]
> For components, what types of things are you looking for exactly?

Something like https://github.com/vadimdemedes/ink 
React components sometimes very good abstractions.
